apps-rendering/src/components/Headline/index.tsx

```diff
@@ -20,7 +20,12 @@ const Headline: React.FC<Props> = ({ item }) => {
 	const format = getFormat(item);
 
 	if (format.display === ArticleDisplay.Immersive) {
-		return <ImmersiveHeadline item={item} />;
+		return (
```

Comment:
I've made an attempt to narrow the props (the domain) of some of these components as much as possible. I'm hoping the fewer things they rely on, the easier they are to reason about and test 🤞.

That said, I've stumbled upon a possible rule of thumb that I'd be interested to hear your thoughts on: where we've broken down a component into sub-components, the `index` component should take `Item` and the sub-components should take only the props they need to render.

I think it's generally good to keep the arguments to a function/props to a component as narrow as possible, but that's difficult in our `index` files because they choose which component to render. This means they have to prop-drill everything that the sub-components need, so they end up with large, unwieldy lists of props. As a compromise I was thinking that we set them up to take `Item`, which should contain almost everything they need. They can then select the much narrower props required by the sub-components (this reminds me a little of `mapStateToProps` in react-redux).